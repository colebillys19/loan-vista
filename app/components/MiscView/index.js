/**
 * MiscView
 * @description ...
 */

import React from 'react';
// import T from 'prop-types';

import MiscEscrow from 'components/MiscEscrow';
import MiscHomeowners from 'components/MiscHomeowners';
import MiscMortgage from 'components/MiscMortgage';
import MiscTaxes from 'components/MiscTaxes';
import TabContainer from 'components/_base-ui/TabContainer';

const MiscView = () => {
  const renderLoading = false;

  return (
    <TabContainer aria-labelledby="misc-tab" id="misc-view">
      <MiscEscrow renderLoading={renderLoading} />
      <MiscTaxes renderLoading={renderLoading} />
      <MiscHomeowners renderLoading={renderLoading} />
      <MiscMortgage renderLoading={renderLoading} />
    </TabContainer>
  );
};

MiscView.propTypes = {
  // error: T.oneOfType([T.bool, T.string]).isRequired,
  // loading: T.bool.isRequired,
  // miscEscrowData: T.array,
  // miscHomeownersData: T.array,
  // miscMortgageData: T.array,
  // miscTaxesData: T.array,
};

export default MiscView;
