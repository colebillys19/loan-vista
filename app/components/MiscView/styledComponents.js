import styled from 'styled-components';

export const MiscViewWrapper = styled.div``;
