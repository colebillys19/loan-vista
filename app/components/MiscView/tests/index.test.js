import React from 'react';
import { render } from 'react-testing-library';

import MiscView from '../index';

import {
  MOCK_ESCROW_DATA as miscEscrowData,
  MOCK_HOMEOWNERS_DATA as miscHomeownersData,
  MOCK_MORTGAGE_DATA as miscMortgageData,
  MOCK_TAXES_DATA as miscTaxesData,
} from './mockData';

const mockProps = {
  error: false,
  loading: false,
  miscEscrowData,
  miscHomeownersData,
  miscMortgageData,
  miscTaxesData,
};

const Component = <MiscView {...mockProps} />;

describe('<MiscView />', () => {
  it('Expect not to log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(Component);
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(Component);
    expect(firstChild).toMatchSnapshot();
  });
});
