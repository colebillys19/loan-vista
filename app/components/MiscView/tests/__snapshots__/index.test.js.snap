// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MiscView /> Should render and match the snapshot 1`] = `
<section
  aria-labelledby="misc-tab"
  class="TabWrapper-j2wsfb-0 beIMbY"
  id="misc-view"
>
  <div
    class="styledComponents__MiscViewWrapper-sc-13d5y5h-0 iNVRfw"
  >
    MiscView
  </div>
</section>
`;
